uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -156,3 +156,15 @@ function hrcore_civicrm_pre($op, $objectName, $objectId, &$params) {
     $currentListener->handle($op, $objectName, $objectId, $params);
   }
 }
+
+/**
+ * Implements hrcore_civicrm_pageRun. This hook adds session variable to
+ * CRM.vars object.
+ *
+ * @link https://docs.civicrm.org/dev/en/master/hooks/hook_civicrm_pageRun/
+ */
+ function hrcore_civicrm_pageRun($page) {
+    CRM_Core_Resources::singleton()->addVars('session', [
+       'contact_id' => CRM_Core_Session::getLoggedInContactID()
+    ]);
+ }
```

Comment:
This hook can actually be used to do a lot more than only add variables to the CRM.vars object. Here are some suggestions:
- Remove the "This hook adds session variable to CRM.vars object" sentence from the docblock
- Extract the `CRM_Core_Resources` call to a separate function that will have the single responsibility of adding the vars. I suggest naming it `_hrcore_set_js_session_vars` (The js is to avoid any confusion with the PHP session), but feel free to use any other name that you might find it's better
- Call this new function from hrcore_civicrm_pageRun